public/scripts/dragdrop.js

```diff
@@ -0,0 +1,107 @@
+import { debounce_timeout } from './constants.js';
+
+/**
+ * Drag and drop handler
+ *
+ * Can be used on any element, enabling drag&drop styling and callback on drop.
+ */
+export class DragAndDropHandler {
+    /** @private @type {JQuery.Selector} */ selector;
+    /** @private @type {(files: File[], event:JQuery.DropEvent<HTMLElement, undefined, any, any>) => void} */ onDropCallback;
+    /** @private @type {NodeJS.Timeout} */ dragLeaveTimeout;
```

Comment:
It's not actually NodeJS timeout, but it's close.

Reply:
Well. The linter docs from Node says it's that class. That's why I took it.
![image](https://github.com/SillyTavern/SillyTavern/assets/9962104/2f9d1508-68f8-409e-9820-4a1db4c4dfe6)
I mostly just wanted to have typesafe checking.

Reply:
Well, it's a `number`. Huh.  
But the NodeJS timer nearly a number too. At least it resolves to numbers as a primitive type. So Cohee was right, it's close to the same.  
I could change the docs to number, but then we'd have to call the function explicitly on window everywhere, or the linter will complain. Dunno if that's worth it.